src/v/cluster/rm_stm_types.h

```diff
@@ -45,22 +45,40 @@ struct expiration_info {
     }
 };
 
-struct transaction_info {
-    enum class status_t { ongoing, preparing, prepared, initiating };
+// Status of a single transaction within a data partition.
+// Not to be confused with user visible transaction status
+// that can potentially span multiple data partitions.
+enum class partition_transaction_status : int8_t {
+    // Data has been appended as a part of the transaction.
+    ongoing,
+    // prepare* are no longer used
+    unused_ignore_preparing,
+    unused_ignore_prepared,
```

Comment:
nit: can we simply remove those and initialize explictly the other two values with `0` and `3`